tests/unit/src/test/scala/tests/decorations/SyntheticDecorationsLspSuite.scala

```diff
@@ -443,6 +443,66 @@ class SyntheticDecorationsLspSuite extends BaseLspSuite("implicits") {
     } yield ()
   }
 
+  test("inferred-type-complex") {
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{
+           |  "a": {}
+           |}
+           |
+           |/a/src/main/scala/Main.scala
+           |import scala.concurrent.ExecutionContextExecutorService
+           |
+           |trait A
+           |trait B
+           |
+           |class Main(implicit ec: ExecutionContextExecutorService){
+           |  // structural types
+           |  val anon1 = new A {}
+           |  val anon2 = new A { def a = 123 }
+           |  val anon3 = new A with B {}
+           |  // existential type
+           |  val job = ec.submit(new Runnable {
+           |     override def run(): Unit = {}
+           |  })
```

Comment:
A minor addition.

I've tried the following sample:
```scala
val runnable = new Runnable {
  override def run(): Unit = {}
}
```
And got printed type:
```scala
val runnable: Object with Runnable {} = new Runnable {
   override def run(): Unit = {}
}
```

Probably in this case, `Object` should be dropped in the same way as it's done for `AnyRef` in `WithType`

Reply:
Good catch, fixed it also!